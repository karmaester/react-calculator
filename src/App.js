import './App.css';
import Big from 'big.js';
import Header from './Header';

function App() {
  return (
    <div className="App">
      <Header title="Math Magicians" />
    </div>
  );
}

export default App;
