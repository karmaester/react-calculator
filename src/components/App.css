body {
  margin: 0;
}

.App {
  margin: auto;
  display: flex;
  flex-direction: column;
}
