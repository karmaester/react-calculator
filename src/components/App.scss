body {
  margin: 0;
  background-color: rgb(253, 223, 123);
}

#root {
	display: flex;
	justify-content: center;
	align-items: center;
	min-height: 100vh;
}

.App {
  margin: auto;
  display: flex;
  flex-direction: column;
}

.display {
  max-width: 600px;
  background-color: gray;
}

.result {
  min-height: 3rem;
  color: white;
  font-size: 2.5rem;
  text-align: end;
  padding-right: 1rem;
}

.btn {
  width: 25%;
  font-size: 2rem;
  padding: 1rem;
}

.btn-large {
  width: 50%;
  font-size: 2rem;
  padding: 1rem;
}

.bg-orange {
  background-color: orange;
}

.bg-gray {
  background-color:rgb(204, 202, 202);
}