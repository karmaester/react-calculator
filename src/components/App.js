/* eslint-disable no-unused-vars */
import './App.css';
import Big from 'big.js';
import Header from './Header';
import Display from './Display';
import ButtonPanel from './ButtonPanel';

function App() {
  const x = new Big(123.4567);
  return (
    <>
      <Header title="Math Magicians" />
      <div className="calculator">
        <Display />
        <ButtonPanel />
      </div>
    </>
  );
}

export default App;
/* eslint-enable no-unused-vars */
