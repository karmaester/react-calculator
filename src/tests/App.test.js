import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../components/App';

test('should render the Calculator Page', () => {
  const component = renderer.create(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
