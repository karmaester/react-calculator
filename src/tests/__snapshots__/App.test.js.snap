// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Calculator Page 1`] = `
<main>
  <div
    className="navbar shadow"
  >
    <div
      className="logo"
    >
      <span>
        Math 
      </span>
      Magician
    </div>
    <div
      className="nav-links"
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        className="nav-link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        className="nav-link"
        href="/quote"
        onClick={[Function]}
      >
        Quote 
      </a>
    </div>
  </div>
  <div
    className="welcome"
  >
    Welcome to Math Magicians
  </div>
</main>
`;
